pkg/restapi/vc/operation/operations.go

```diff
@@ -1324,6 +1325,115 @@ func (o *Operation) parseAndVerifyVP(vpBytes []byte) error {
 	return nil
 }
 
+func (o *Operation) queryVault(vaultID, vcID string) ([]string, error) {
+	vcIDMAC, err := o.macCrypto.ComputeMAC([]byte(vcID), o.macKeyHandle)
+	if err != nil {
+		return nil, err
+	}
+
+	vcIDIndexValueEncoded := base64.URLEncoding.EncodeToString(vcIDMAC)
+
+	return o.edvClient.QueryVault(vaultID, &models.Query{
+		Name:  o.vcIDIndexNameEncoded,
+		Value: vcIDIndexValueEncoded,
+	})
+}
+
+func (o *Operation) retrieveCredential(rw http.ResponseWriter, profileName string, docURLs []string) {
+	var retrievedVC []byte
+
+	switch len(docURLs) {
+	case 0:
+		o.writeErrorResponse(rw, http.StatusBadRequest,
+			fmt.Sprintf(`no VC under profile "%s" was found with the given id`, profileName))
+	case 1:
+		docID := getDocIDFromURL(docURLs[0])
+
+		var err error
+
+		retrievedVC, err = o.retrieveVC(profileName, docID, "retrieving VC")
+		if err != nil {
+			o.writeErrorResponse(rw, http.StatusInternalServerError, err.Error())
+
+			return
+		}
+	default:
+		// Multiple VCs were found with the same id. This is technically possible under the right circumstances
+		// when storing the same VC multiples times in a store provider that follows an "eventually consistent"
+		// consistency model. If they are all the same, then just return the first one arbitrarily.
+		// ToDo: If the multiple VCs with the same ID all are identical then delete the extras and leave only one.
+		// https://github.com/trustbloc/edge-service/issues/262
+		var err error
+
+		retrievedVC, err = o.verifyMultipleMatchingVCsAreIdentical(profileName, docURLs)
+		if err != nil {
+			o.writeErrorResponse(rw, http.StatusInternalServerError, err.Error())
```

Comment:
I don't think internal server error is appropriate here. We could maybe use 409 - CONFLICT: 

> The request could not be completed due to a conflict with the current state of the target resource. This code is used in situations where the user might be able to resolve the conflict and resubmit the request.

Reply:
Note that the `verifyMultipleMatchingVCsAreIdentical` method can also return an error if the `retrieveVC` method fails, and I think failures from `retrieveVC` are still internal server errors. So I've made `verifyMultipleMatchingVCsAreIdentical` return a status code now depending on which failure - if there are multiple VCs with differing contents, I return the 409 Conflict error now. Let me know if you disagree with my change.